banyand/internal/storage/segment.go

```diff
@@ -160,6 +167,20 @@ func (sc *segmentController[T, O]) selectTSTables(timeRange timestamp.TimeRange)
 }
 
 func (sc *segmentController[T, O]) createTSTable(ts time.Time) (TSTableWrapper[T], error) {
+	data := []byte(currentVersion)
+	versionPath := filepath.Join(sc.location, versionFilename)
+	fileSystem := fs.NewLocalFileSystem()
+	lf, err := fileSystem.CreateLockFile(versionPath, filePermission)
```

Comment:
If the version file exists, the segment controller should check whether the version is compatible.